src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -230,24 +230,36 @@
    line-number :- schema/Int
    original-message :- schema/Str]
   (IllegalArgumentException.
-   (format (str "Problem loading service '%s' on line '%s' in bootstrap "
-                "configuration file '%s':\n%s")
-           entry line-number bootstrap-file original-message)))
+   (format (str "%s:%s\nProblem loading service '%s':\n%s")
+           bootstrap-file line-number entry original-message)))
+
+(schema/defn resolve-and-handle-errors! :- (schema/maybe (schema/protocol services/ServiceDefinition))
+  "Attemps to resolve a bootstrap entry into a ServiceDefinition.
+  If the bootstrap entry can't be resolved, logs a warning and returns nil.
+
+  Throws an IllegalArgumentException if there is a problem parsing the bootstrap
+  entry, or if the service is found but it has an invalid service graph."
+  [{:keys [bootstrap-file line-number entry] :as bootstrap-entry} :- AnnotatedBootstrapEntry]
+  (try+
+    (let [{:keys [namespace service-name]} (parse-bootstrap-line! entry)]
+      (resolve-service! namespace service-name))
+    (catch [:type ::missing-service] {:keys [message]}
+      (log/warnf "%s:%s\nUnable to load service '%s'" bootstrap-file line-number entry))
```

Comment:
With the current code, I get error messages like this when there's a problem, which I find a little hard to read:

```
2016-04-18 15:55:34,576 WARN  [p.t.bootstrap] ./examples/java_service/bootstrap.cfg:2
Unable to load service 'crappy/service'
```

I find it a little more readable to have the error message lead off with "Unable to load service ... " and have the service name, bootstrap file, and line number follow.  Having it all one line would seem reasonable to me as well.